sympy/combinatorics/pc_groups.py

```diff
@@ -0,0 +1,441 @@
+from sympy.core import Basic
+from sympy import sieve
+from sympy.combinatorics.perm_groups import PermutationGroup
+from sympy.printing.defaults import DefaultPrinting
+
+class PolycyclicGroup(DefaultPrinting):
+
+    is_group = True
+    is_solvable = True
+
+    def __init__(self, pc_sequence, pc_series):
+        self.pcgs = pc_sequence
+        self.pc_series = pc_series
+
+    def relative_order(self):
+        rel_orders = []
+        for i in range(len(self.pc_series)-1):
+            G = self.pc_series[i]
+            H = self.pc_series[i+1]
+            rel_orders.append(G.order() // H.order())
+        return rel_orders
+
+    def is_prime_order(self):
+        for order in self.relative_order():
+            if order not in sieve:
+                return False
+        return True
+
+    def length(self):
+        return len(self.pcgs)
+
+    def exponent_vector(self, element, group):
+        G = PermutationGroup()
+        for g in self.pcgs:
+            G = PermutationGroup(G.generators + [g])
+        gens = G.generator_product(element, original = True)
+
+        perm_to_free = {}
+        for sym, g in zip(group.generators, self.pcgs):
+            perm_to_free[g] = sym
+        w = group.identity
+        for g in gens:
+            w = w*perm_to_free[g]
+
+        pc_presentation = self.pc_presentation(group)
+        collector = Collector(pc_presentation, self.relative_order(), group)
+        word = collector.collected_word(w)
+
+        index = {s: i for i, s in enumerate(group.symbols)}
+        exp_vector = [0]*len(group)
+        word = word.array_form
+        for t in word:
+            exp_vector[index[t[0]]] = t[1]
+        return exp_vector
+
+    def power_relations(self, group):
+        """
+        Return a list of power relations.
+        If `x` is a free group element and `re` represents
+        the relative order of `x` then power relation of a
+        single generator `x` is `x**re`.
+
+        Examples
+        ========
+        >>> from sympy.combinatorics.pc_groups import PolycyclicGroup
+        >>> from sympy.combinatorics.named_groups import SymmetricGroup
+        >>> from sympy.combinatorics.free_groups import free_group
+        >>> F, x0, x1 = free_group("x0, x1")
+        >>> G = SymmetricGroup(4)
+        >>> pc_group = G.polycyclic_group()
+        >>> group = F
+        >>> pc_group.power_relations(group)
+        [x1**3, x0**2]
```

Comment:
These are the lhs expressions of power relations. There should also be the rhs expressions in normal form.

Reply:
We are only computing lhs in both the methods(this one and below) which can be directly feed to the `pc_presentation` where rhs is computed may be we can make both the methods private with different method names something like `conjugate_lhs` and `power_lhs`.
This is just to reduce the number of lines in method `pc_presentation`!